src/wallet/wallet2.cpp

```diff
@@ -2197,6 +2334,9 @@ crypto::secret_key wallet2::generate(const std::string& wallet_, const std::stri
   generate_genesis(b);
   m_blockchain.push_back(get_block_hash(b));
 
+  m_subaddress_labels.clear();
+  add_subaddress_account(tr("Primary account"));
+  m_current_subaddress_account = 0;
```

Comment:
As @moneromooo-monero clarified that `rescan_bc` is rather discouraged to be used often, the clearing of subaddress hash table and labels is moved to `wallet2::clear()`.